modules/cudaarithm/src/cuda/polar_cart.cu

```diff
@@ -116,6 +116,25 @@ void cv::cuda::phase(InputArray _x, InputArray _y, OutputArray _dst, bool angleI
     syncOutput(dst, _dst, stream);
 }
 
+void cv::cuda::phase(InputArray _xy, OutputArray _dst, bool angleInDegrees, Stream& stream)
+{
+    GpuMat xy = getInputMat(_xy, stream);
+
+    CV_Assert( xy.type() == CV_32FC2 );
+
+    GpuMat dst = getOutputMat(_dst, xy.size(), CV_32FC1, stream);
+
+    GpuMat_<float2> xyc(xy.reshape(2));
```

Comment:
I know the existing functions use reshape to convert a `GpuMat` to a `GpuMat_` before being passed to `gridTransformxxx` but I find this confusing because nothing is being reshaped, would it not be better to use `globPtr<>` directly on the `GpuMats`. e.g.

```
    if (angleInDegrees)
        gridTransformUnary(globPtr<float2>(xy), globPtr<float>(dst), direction_interleaved_func<float2, true>(), stream);
```

If so the existing routines could be updated to remove the bloat.